ts-client/src/amm/index.ts

```diff
@@ -0,0 +1,859 @@
+
+    const adminTokenFee =
+      this.tokenA.address === inTokenMint.toBase58() ? this.poolState.adminTokenAFee : this.poolState.adminTokenBFee;
+
+    let preInstructions: Array<TransactionInstruction> = [];
+    const [[userSourceToken, createUserSourceIx], [userDestinationToken, createUserDestinationIx]] =
+      await this.createATAPreInstructions(owner, [sourceToken, destinationToken]);
+
+    createUserSourceIx && preInstructions.push(createUserSourceIx);
+    createUserDestinationIx && preInstructions.push(createUserDestinationIx);
+
+    if (sourceToken.equals(WRAPPED_SOL_MINT)) {
+      preInstructions = preInstructions.concat(wrapSOLInstruction(owner, userSourceToken, inAmountLamport.toNumber()));
+    }
+
+    preInstructions.push(await this.getApySyncInstructions());
+
+    const postInstructions: Array<TransactionInstruction> = [];
+    if (WRAPPED_SOL_MINT.equals(destinationToken)) {
+      const unwrapSOLIx = await unwrapSOLInstruction(owner);
+      unwrapSOLIx && postInstructions.push(unwrapSOLIx);
+    }
+
+    const swapTx = await this.program.methods
+      .swap(inAmountLamport, outAmountLamport)
+      .accounts({
+        aTokenVault: this.vaultA.vaultState.tokenVault,
+        bTokenVault: this.vaultB.vaultState.tokenVault,
+        aVault: this.poolState.aVault,
+        bVault: this.poolState.bVault,
+        aVaultLp: this.poolState.aVaultLp,
+        bVaultLp: this.poolState.bVaultLp,
+        aVaultLpMint: this.vaultA.vaultState.lpMint,
+        bVaultLpMint: this.vaultB.vaultState.lpMint,
+        userSourceToken,
+        userDestinationToken,
+        user: owner,
+        adminTokenFee,
+        pool: this.address,
+        tokenProgram: TOKEN_PROGRAM_ID,
+        vaultProgram: this.vaultProgram.programId,
+      })
+      .remainingAccounts(getRemainingAccounts(this.poolState))
+      .preInstructions(preInstructions)
+      .postInstructions(postInstructions)
+      .transaction();
+
+    return new Transaction({
+      feePayer: owner,
+      ...(await this.program.provider.connection.getLatestBlockhash('finalized')),
+    }).add(swapTx);
+  }
+
+  public async getDepositQuote(
+    tokenAInAmount: BN,
+    tokenBInAmount: BN,
+    isImbalance = false,
+    slippage?: number,
+  ): Promise<{
+    poolTokenAmountOut: BN;
+    tokenAInAmount: BN;
+    tokenBInAmount: BN;
+  }> {
+    const slippageRate = slippage ?? DEFAULT_SLIPPAGE;
+    const { tokenAAmount, tokenBAmount } = await this.getPoolInfo();
+
+    const { poolLpSupply, vaultALpSupply, vaultBLpSupply, poolVaultALp, poolVaultBLp } = await this.getSplInfo();
+
+    const vaultAWithdrawableAmount = await this.vaultA.getWithdrawableAmount();
+    const vaultBWithdrawableAmount = await this.vaultB.getWithdrawableAmount();
+
+    // Constant product pool
+    if (tokenAInAmount.eq(new BN(0)) && !isImbalance) {
+      const poolTokenAmountOut = this.getShareByAmount(tokenBInAmount, tokenBAmount, poolLpSupply);
+
+      if (this.isStablePool) {
```

Comment:
When it's balanced deposit, the calculation for constant_product, and stable pool are the same